### PR TITLE
[FIX] website: fix configurator/edit menu ui

### PR DESCRIPTION
**Pull request by younn-o**
[FIX] website: fix edit menu dialog ui

This commit adds the hint that [1] missed when adapting the "Edit menu"
dialog to owl, and adds a margin removed by the bootsrap update.

[1]: https://github.com/odoo/odoo/commit/31cc10b91dc7762e23b4bde9b945be0c4ce3fe3b

task-2687506

-----

[FIX] website: fix configurator style

When [1] moved the configurator frontend to a backend client action, it
did not correctly adapt the style: the backend's root element has a
font-size of 12px when the frontend has the default one (16px). As a
result, every css property using rem units was inaccurate.
This commit temporarily sets the root element font size to 16px when the
configurator client action is displayed to restore the style it had on
versions 15.X.
It also adapts the style to bootsrap 5.

[1]: https://github.com/odoo/odoo/commit/31cc10b91dc7762e23b4bde9b945be0c4ce3fe3b

task-2687506

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by younn-o** [comment]
@qsm-odoo Here is a lazy solution to restore the configurator's previous style, without changing all classes depending on rem units. What do you think about it ?

File: addons/website/static/src/client_actions/configurator/configurator.js [hunk 1]
```diff
@@ -627,6 +627,18 @@ class Configurator extends Component {
...
+            const rootEl = document.querySelector(':root');
+            const initalFontSize = rootEl.style.fontSize;
+            rootEl.style.fontSize = '16px';
+            return () => {
+                rootEl.style.fontSize = initalFontSize;
+                if (!rootEl.getAttribute('style')) {
+                    rootEl.removeAttribute('style');
+                }
+            };
+        }, () => []);
```
Comment (qsm-odoo) [inline addons/website/static/src/client_actions/configurator/configurator.js:641]: No, instead the CSS has to be adapted I guess. It's also fine that the design changed a little to match a bit more the backend spirit IMO. I think it's not a commit for you to make in the end, a designer should review the configurator design post all refactorings.

File: addons/website/static/src/client_actions/configurator/configurator.scss [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+        color: $o-warning !important;
+    }
+
+    .card {
+        background-color: $white !important;
+
+        .card-body {
+            background-color: rgba($white, 0.9) !important;
+        }
+    }
```
Comment (qsm-odoo) [inline addons/website/static/src/client_actions/configurator/configurator.scss:34]: I don't see why the bootstrap behavior would have to be reviewed for the configurator. At that point, using bootstrap make no sense if we don't even like what it does and override everything :grimacing: Again, I think a design should review the structure etc. I'll remove this commit and create a task for it.

File: addons/website/static/src/components/dialog/dialog.scss [hunk 3]
```diff
@@ -2,4 +2,8 @@
     label {
         font-weight: $font-weight-bold;
     }
+
+    .form-group {
+        margin-bottom: 1rem;
+    }
```
Comment (qsm-odoo) [inline addons/website/static/src/components/dialog/dialog.scss:8]: The form-group class apparently does not exist anymore in BS5... and apparently the class was kept everywhere in the codebase anyway :( Migrating to BS5 does not mean restoring its old rules case by case... :grimacing: Here what you have to do is properly migrate to BS5, not restore BS4 rules to make it work :/

File: addons/website/static/src/components/dialog/edit_menu.xml [hunk 4]
```diff
@@ -16,6 +16,7 @@
             </label>
             <div class="col-md-9">
                 <input id="url_input" t-ref="url-input" type="text" t-model="url.input.value" class="form-control" t-att-class="{ 'is-invalid': url.input.hasError }"/>
+                <small class="form-text text-muted">Hint: Type '/' to search an existing page and '#' to link to an anchor.</small>
```
Comment (qsm-odoo) [inline addons/website/static/src/components/dialog/edit_menu.xml:19]: I'll move this to another PR and keep this PR open in draft, we'll see what to do with it when you are back

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/src/components/dialog/edit_menu.xml (hunk 4 above)
Comment (qsm-odoo) [reply addons/website/static/src/components/dialog/edit_menu.xml:19]: https://github.com/odoo/odoo/pull/97733 -> also removed the text-muted class which had no effect

**Comment by qsm-odoo** [comment]
Design was adapted another way.

**Pull request closed without merging**